libres/lib/enkf/local_updatestep.cpp

```diff
@@ -65,19 +62,18 @@ void local_updatestep_free__(void *arg) {
 
 void local_updatestep_add_ministep(local_updatestep_type *updatestep,
                                    local_ministep_type *ministep) {
-    vector_append_ref(
-        updatestep->ministep,
-        ministep); /* Observe that the vector takes NO ownership */
+    updatestep->ministep.emplace_back(
+        std::unique_ptr<local_ministep_type>(ministep));
```

Comment:
The previous code has the comment "Observe that the vector takes NO ownership". `std::unique_ptr` takes ownership of its pointer.

Most likely, you should just change `ministep` to be `std::vector<local_ministep_type*>`. This is not ideal (raw pointers should be avoided in good modern C++), but it's a step in the right direction.

Reply:
You didn't make the vector the owner of the pointer (it is that already). You made it the owner of the object pointed to by the pointer. Those objects are owned by `local_config_type` and are deleted by `local_config_free`. You're getting a segfault with `delete` because of this (double free).